### ISSUE TITLE
Avoid libbpf-dev dependency for CNDP applications

### ISSUE THREAD

**Issue by manojgop** [CloudNativeDataPlane/cndp#226]
Currently CNDP applications using `xskdev.h` header needs to include `ibbpf-dev` as dependency because `xskdev.h` includes `<bpf/xsk.h>`. Is it possible to refactor `xskdev_info_t` structure in `xskdev.h`  such a way that we can include “xskdev_xxx” data types in some private structure defined in an internal header file like `xskdev_internal.h` and then have a member variable like `void *priv` in `xskdev_info_t` structure.

File: xskdev_internal.h
===============================
```
#include <bpf/xsk.h>

typedef struct xskdev_info_internal {
    xskdev_rxq_t rxq;              /**< RX queue */
    xskdev_txq_t txq;              /**< TX queue */
    xskdev_get_mbuf_addr_tx_t     __get_mbuf_addr_tx;         /**< Internal function to set the mbuf address on tx */
    xskdev_get_mbuf_rx_t           __get_mbuf_rx;                   /**< Internal function to get the mbuf address on rx */
    xskdev_pull_cq_addr_t          __pull_cq_addr;                  /**< Internal function to pull the complete queue */
}
```

File: xskdev.h
=================
```
typedef struct xskdev_info {
      …..

      void *priv; /**  >struct xskdev_info_internal > **/
       …..
}
```

File: xskdev.c
=================
```
#include  xskdev_internal.h
#include xskdev.h

xskdev_info  xskdev;
struct xskdev_info_internal *xskdev_internal = (struct xskdev_info_internal *) xskdev->priv;
```

**Comment by manojgop** [comment]
> I don't understand what the benefit of this is? that dependency is based on the low level APIs/structs living in libbpf?
> 
> Also the location of the xsk.h file is dependent on kernel version as it may come from libxdp?

Can we hide libbpf-dev dependency from CNDP applications ? That way CNDP applications doesn't have to include libbpf-dev as a dependency. CNDP apps only would need libbpf library dependency and not header files. One option I'm thinking is the possibility to include "bpf/xsk.h" in xskdev.c instead of xskdev.h .

**Comment by maryamtahhan** [comment]
If we strongly feel we need to change it we can do it. However from this point onwards I would resist changing "core" APIs unless these changes are really needed.

**Comment by KeithWiles** [comment]
I agree with Maryam here. I do not see the need to not include libbpf-dev in the install process. Having too many void * pointer can make it difficult for the compiler to detect issues. I am worried we are making this change just for Rust and after the PMD static builds causing a lot of issues, it does not seem like it provides significant benefit overall.

We can leave the issue as an open for now.

**Comment by manojgop** [comment]
Issue was related to including libbpf-dev for CNDP OMEC-UPF / BESS instead of just using libbpf0. I think this header file dependency could be resolved. But if it's too much of a hassle we can deprioritize or close this issue.

**Comment by manojgop** [comment]
I'll try to convince the OMEC UPF maintainers that libbpf-dev is a required dependency for CNDP since CNDP public header file (xskdev.h) includes libbpf header file. If still there is concern we can consider refactoring xskdev.h.

**Comment by maryamtahhan** [comment]
@manojgop Can we close this issue now?
